staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -137,6 +137,12 @@ func (f *defaultPortForwarder) ForwardPorts(method string, url *url.URL, opts Po
 		return err
 	}
 	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: transport}, method, url)
+	if cmdutil.PortForwardWebsockets.IsEnabled() {
+		dialer, err = portforward.NewTunnelingDialer(url, opts.Config, opts.Ports)
```

Comment:
tunneling dialer probably doesn't need opts.Ports

if we fail to create/upgrade the websocket request, we'll need to fall back to the standard spdy dialer, right? that probably means we want a fallback dialer, something like:

```
spdyDialer := ...
if cmdutil.PortForwardWebsockets.IsEnabled() {
  websocketDialer := ...
  dialer = fallbackDialer(websocketDialer, spdyDialer)
} else {
  dialer = spdyDialer
}
```

Reply:
Added a `fallbackDialer` and unit tests. Removed `opts.Ports`. Done.